lib/src/directory_watcher/linux.dart

```diff
@@ -47,8 +47,8 @@ class _LinuxDirectoryWatcher
   bool get isReady => _readyCompleter.isCompleted;
 
   @override
-  Future get ready => _readyCompleter.future;
-  final _readyCompleter = Completer();
+  Future<void> get ready => _readyCompleter.future;
```

Comment:
Technically, this is breaking, because void values are not assignable.

It might be fine though in practice? But there is a bit of risk involved.

Reply:
I originally changed the type on the public `Watcher` interface too, but didn't end up making that change for this reason. 

Shouldn't it be non-breaking in this case since it's a private implementation of it?

My understanding was that `await watcher.ready` should still be assignable and return `null` since the public return type is still `Future`. Perhaps I'm misunderstanding though?